### PR TITLE
formula_creator: update go build args

### PR DESCRIPTION
**Pull request by dawidd6**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

We started using `"-ldflags", "-s -w", "-trimpath"` arguments passed to `go build` and it would be nice to have them added to new formulae too by `brew create --go`.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> We started using `"-ldflags", "-s -w", "-trimpath"` arguments passed to `go build`

When was that and what do they do? Apologies for being late to the party!

**Comment by dawidd6** [comment]
https://github.com/Homebrew/homebrew-core/issues/47627#issuecomment-563071225

**Comment by MikeMcQuaid** [comment]
Thanks! What do you think about using the `std_go_args` discussed there instead?

**Comment by dawidd6** [comment]
I don't think it's a good idea. At least with those `ldflags` options, because some formulae need to set some additional `ldflags`, like a version variable at build time and then those `ldflags` defined earlier on command line are overwritten by the latter. So those formulae wouldn't use `std_go_args` and instead define them manually, which defeats the whole idea in my opinion.

**Comment by MikeMcQuaid** [comment]
> So those formulae wouldn't use `std_go_args` and instead define them manually, which defeats the whole idea in my opinion.

I don't think having a `std_go_args` not used by all Go formulae would make this not useful to those that don't do that. @Homebrew/core thoughts?

**Comment by bayandin** [comment]
We could have something `std_go_ldflags`, which will be easy to copy and extend.

**Comment by dawidd6** [comment]
`std_go_ldflags` wouldn't cover `-trimpath` though, cause it is not an `ldflags` option. `std_go_args` might be alright then.

File: Library/Homebrew/formula_creator.rb [hunk 1]
```diff
@@ -137,7 +137,7 @@ def install
                                   "--disable-silent-rules",
                                   "--prefix=\#{prefix}"
         <% elsif mode == :go %>
-            system "go", "build", "-o", "\#{bin}/\#{name}"
+            system "go", "build", "-o", bin/name, *std_go_args
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_creator.rb]: maybe even `-o bin/name` could be part of `std_go_args`? What if you specify multiple `-o`?

Comment (dawidd6) [reply Library/Homebrew/formula_creator.rb]: Only the latter option specified is used.

I updated this PR and the other one accordingly.

**Comment by MikeMcQuaid** [comment]
Thanks again, again @dawidd6!

**Pull request merged**